src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -1378,9 +1378,11 @@ public char FirstCharOfOneOrMulti()
             return Kind == RegexNodeKind.One ? Ch : Str![0];
         }
 
-        /// <summary>Finds the guaranteed beginning character of the node, or null if none exists.</summary>
-        public (char Char, string? String)? FindStartingCharacterOrString()
+        /// <summary>Finds the guaranteed beginning literal(s) of the node, or null if none exists.</summary>
+        public (char Char, string? String, string? SetChars)? FindStartingLiteral(int maxSetCharacters = 5) // 5 is max optimized by IndexOfAny today
```

Comment:
Might be a dumb question, but why do we limit it to 5 characters?

Reply:
You mean why does IndexOfAny limit it to 5?  Diminishing returns.  Technically now the comment is a bit incorrect; I made a change last week so that span.IndexOfAny will use the same probabilistic map feature that string.IndexOfAny has been using, and that kicks in starting at 6+.  But <= 5 are vectorized, which is where the real bump here comes from.